pyproject.toml

```diff
@@ -122,6 +124,7 @@ all = [
     "pydantic",
     "schema-enforcer",
     "termcolor",
+    "slurpit",
```

Comment:
This should be `slurpit-sdk` to match what is in the dependencies.  Then you can have a group called `slurpit` just like the `aci`, `aristacv` or `sensible`

```
slurp = [
    "slurpit-sdk",
]
```

Reply:
Fixed this in a new commit. However there is a conflict when updating the poetry lock file due to IP Fabric and Slurpit using httpx

```
Updating dependencies
Resolving dependencies... (20.2s)

The current project's supported Python range (>=3.8,<3.13) is not compatible with some of the required packages Python requirement:
  - slurpit-sdk requires Python <4.0,>=3.10, so it will not be satisfied for Python >=3.8,<3.10

Because no versions of ipfabric match >6.6.2,<6.6.3b0 || >6.6.3b0,<6.6.3b1 || >6.6.3b1,<6.6.3b2 || >6.6.3b2,<6.6.3b3 || >6.6.3b3,<6.6.3b4 || >6.6.3b4,<6.6.3b5 || >6.6.3b5,<6.6.3b6 || >6.6.3b6,<6.6.4 || >6.6.4,<6.7.0
 and ipfabric (6.6.3b0) depends on httpx (>=0.26,<0.27), ipfabric (>6.6.2,<6.6.3b1 || >6.6.3b1,<6.6.3b2 || >6.6.3b2,<6.6.3b3 || >6.6.3b3,<6.6.3b4 || >6.6.3b4,<6.6.3b5 || >6.6.3b5,<6.6.3b6 || >6.6.3b6,<6.6.4 || >6.6.4,<6.7.0) requires httpx (>=0.26,<0.27).
And because ipfabric (6.6.3b1) depends on httpx (>=0.26,<0.27), ipfabric (>6.6.2,<6.6.3b2 || >6.6.3b2,<6.6.3b3 || >6.6.3b3,<6.6.3b4 || >6.6.3b4,<6.6.3b5 || >6.6.3b5,<6.6.3b6 || >6.6.3b6,<6.6.4 || >6.6.4,<6.7.0) requires httpx (>=0.26,<0.27).
And because ipfabric (6.6.3b2) depends on httpx (>=0.26,<0.27)
 and ipfabric (6.6.3b3) depends on httpx (>=0.26,<0.27), ipfabric (>6.6.2,<6.6.3b4 || >6.6.3b4,<6.6.3b5 || >6.6.3b5,<6.6.3b6 || >6.6.3b6,<6.6.4 || >6.6.4,<6.7.0) requires httpx (>=0.26,<0.27).
And because ipfabric (6.6.3b4) depends on httpx (>=0.26,<0.27)
 and ipfabric (6.6.3b5) depends on httpx (>=0.26,<0.27), ipfabric (>6.6.2,<6.6.3b6 || >6.6.3b6,<6.6.4 || >6.6.4,<6.7.0) requires httpx (>=0.26,<0.27).
And because ipfabric (6.6.3b6) depends on httpx (>=0.26,<0.27)
 and ipfabric (6.6.2) depends on httpx (>=0.26,<0.27), ipfabric (>=6.6.2,<6.6.4 || >6.6.4,<6.7.0) requires httpx (>=0.26,<0.27).
And because ipfabric (6.6.4) depends on httpx (>=0.26,<0.27)
 and slurpit-sdk (0.9.58) depends on httpx (>=0.27.0,<0.28.0), ipfabric (>=6.6.2,<6.7.0) is incompatible with slurpit-sdk (0.9.58).
Because no versions of slurpit-sdk match >0.9.55,<0.9.58 || >0.9.58
 and slurpit-sdk (0.9.55) requires Python <4.0,>=3.10, slurpit-sdk is forbidden.
Thus, ipfabric (>=6.6.2,<6.7.0) is incompatible with slurpit-sdk (>=0.9.55).
So, because nautobot-chatops depends on both ipfabric (~6.6.2) and slurpit-sdk (>=0.9.55), version solving failed.

  • Check your dependencies Python requirement: The Python requirement can be specified via the `python` or `markers` properties
    
    For slurpit-sdk, a possible solution would be to set the `python` property to ">=3.10,<3.13"```
    
   I cant pin slurpit for 3.10 onwards due to Nautobot testing against 3.8.